askama/src/lib.rs

```diff
@@ -502,6 +502,10 @@ pub trait Template {
     fn extension() -> Option<&'static str>
     where
         Self: Sized;
+    fn dyn_size_hint(&self) -> usize;
+}
+
+pub trait TemplateSizeHint {
```

Comment:
Let's rename this to `SizedTemplate` and move both `extension()` and `size_hint()` into it?